drivers/flash/flash_stm32_ospi.c

```diff
@@ -0,0 +1,1161 @@
+	sCommand.InstructionSize = HAL_OSPI_INSTRUCTION_8_BITS;
+	sCommand.InstructionDtrMode = HAL_OSPI_INSTRUCTION_DTR_DISABLE;
+	sCommand.Instruction = SPI_NOR_CMD_CFGREG2;
+	sCommand.Address = SPI_NOR_REG2_ADDR3;
+	sCommand.AddressMode = HAL_OSPI_ADDRESS_1_LINE;
+	sCommand.AddressSize = HAL_OSPI_ADDRESS_32_BITS;
+	sCommand.AddressDtrMode = HAL_OSPI_ADDRESS_DTR_DISABLE;
+	sCommand.AlternateBytesMode = HAL_OSPI_ALTERNATE_BYTES_NONE;
+	sCommand.DataMode = HAL_OSPI_DATA_1_LINE;
+	sCommand.DataDtrMode= HAL_OSPI_DATA_DTR_DISABLE;
+	sCommand.NbData = 1;
+	sCommand.DummyCycles = 0;
+	sCommand.DQSMode = HAL_OSPI_DQS_DISABLE;
+	sCommand.SIOOMode = HAL_OSPI_SIOO_INST_EVERY_CMD;
+
+	if (HAL_OSPI_Command(hospi, &sCommand, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI write op failed");
+		printk("OSPI write op failed\n");
+		return -EIO;
+	}
+	/* Write Configuration register 2 with new dummy cycles */
+	tmp = SPI_NOR_CR2_DUMMY_CYCLES_66MHZ;
+	if (HAL_OSPI_Transmit(hospi, &tmp, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI write dummy failed");
+		printk("OSPI write dummy failed\n");
+		return -EIO;
+	}
+	if (flash_stm32_ospi_autoPolling_WIP(hospi) != 0) {
+		LOG_ERR("OSPI write dummy failed");
+		printk("OSPI write dummy failed\n");
+		return -EIO;
+	}
+	/* Enable writing to memory in order to set Octal DTR mode */
+	if (flash_stm32_ospi_write_enable(hospi) != 0) {
+		LOG_ERR("OSPI write dummy failed");
+		printk("OSPI write dummy failed\n");
+		return -EIO;
+	}
+	/* Initialize OCTOSPI to Indirect write mode to configure Octal mode */
+	sCommand.Instruction = SPI_NOR_CMD_CFGREG2;
+	sCommand.Address = SPI_NOR_REG2_ADDR1;
+	sCommand.AddressMode = HAL_OSPI_ADDRESS_1_LINE;
+	if (HAL_OSPI_Command(hospi, &sCommand, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI write dummy failed");
+		printk("OSPI write dummy failed\n");
+		return -EIO;
+	}
+	/* Write Configuration register 2 with with Octal mode */
+	tmp = SPI_NOR_CR2_DTR_OPI_EN;
+	if (HAL_OSPI_Transmit(hospi, &tmp, HAL_OSPI_TIMEOUT_DEFAULT_VALUE) != HAL_OK) {
+		LOG_ERR("OSPI Configuration register 2 failed");
+		printk("OSPI Configuration register 2 failed\n");
+		return -EIO;
+	}
+	return 0;
+}
+
+static int flash_stm32_ospi_init(const struct device *dev)
+{
+	const struct flash_stm32_ospi_config *dev_cfg = DEV_CFG(dev);
+	struct flash_stm32_ospi_data *dev_data = DEV_DATA(dev);
+	uint32_t ahb_clock_freq;
+	uint32_t prescaler = 0;
+	int ret;
+
+	/* Signals configuration */
+	ret = pinctrl_apply_state(dev_cfg->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		LOG_ERR("OSPI pinctrl setup failed (%d)", ret);
+		return ret;
+	}
+
+	/* Initializes the independent peripherals clock */
+//	__HAL_RCC_OSPI_CONFIG(RCC_OSPICLKSOURCE_PLL1); /* PLL1 is the clock source */
+	__HAL_RCC_OSPI_CONFIG(RCC_OSPICLKSOURCE_SYSCLK); /* */
+	__HAL_RCC_OSPI2_CLK_ENABLE();
+	__HAL_RCC_OSPI2_FORCE_RESET();
+	__HAL_RCC_OSPI2_RELEASE_RESET();
+
```

Comment:
Clock related stuff is supposed to be here, or is it just here for quick testing?

Reply:
yes,
Done now by 
```
	if (clock_control_on(DEVICE_DT_GET(STM32_CLOCK_CONTROL_NODE),
			     (clock_control_subsys_t) &dev_cfg->pclken) != 0) {
		LOG_DBG("Could not enable OSPI clock");
		return -EIO;
	}
```